lib/ABI/FunctionType/Layout.cpp

```diff
@@ -0,0 +1,963 @@
+    }
+  }
+
+  return Result;
+}
+
+std::pair<DistributedArguments, std::size_t>
+ToRawConverter::distributeImpl(model::QualifiedType Type,
+                               RegisterSpan Registers,
+                               std::size_t OccupiedRegisterCount,
+                               std::size_t AllowedRegisterLimit,
+                               bool AllowPuttingPartOfAnArgumentOnStack) const {
+  DistributedArguments Result;
+
+  LoggerIndent Indentation(Log);
+  revng_log(Log, "Distributing a value between the registers and the stack.");
+  revng_log(Log,
+            OccupiedRegisterCount << " registers are already occupied, while "
+                                     "only "
+                                  << AllowedRegisterLimit
+                                  << " registers are available to be used.");
+  revng_log(Log, "The total number of registers is " << Registers.size());
+
+  std::size_t Size = *Type.size();
+  std::size_t Alignment = *ABI.alignment(Type);
+  revng_log(Log, "The type:\n" << serializeToString(Type));
+  revng_log(Log,
+            "It's size is " << Size << " and it's alignment is " << Alignment
+                            << ".");
+
+  // Precompute the last register allowed to be used.
+  std::size_t LastRegister = OccupiedRegisterCount + AllowedRegisterLimit;
+
+  // Define the counters: one for the number of registers the current value
+  // could occupy, and one for the total size of said registers.
+  std::size_t ConsideredRegisterCounter = OccupiedRegisterCount;
+  std::size_t SizeCounter = 0;
+
+  // Keep adding the registers until either the total size exceeds needed or
+  // we run out of allowed registers.
+  if (!Registers.empty()) {
+    auto ShouldContinue = [&]() -> bool {
+      return SizeCounter < Size && ConsideredRegisterCounter < Registers.size()
+             && ConsideredRegisterCounter < LastRegister;
+    };
+    while (ShouldContinue()) {
+      const auto CurrentRegister = Registers[ConsideredRegisterCounter++];
+      SizeCounter += model::Register::getSize(CurrentRegister);
+    }
+  }
+
+  if (SizeCounter >= Size) {
+    if (ConsideredRegisterCounter - OccupiedRegisterCount == 1) {
+      revng_log(Log, "A single register is sufficient to hold the value.");
+    } else {
+      revng_log(Log,
+                (ConsideredRegisterCounter - OccupiedRegisterCount)
+                  << " registers are sufficient to hold the value.");
+    }
+  }
+
+  // Take the alignment into consideration on the architectures that require
+  // padding to be inserted even for arguments passed in registers.
+  std::optional<std::size_t> PaddingRegisterIndex = std::nullopt;
+  if (ABI.OnlyStartDoubleArgumentsFromAnEvenRegister()) {
+    if (Size >= ABI.getPointerSize() && Alignment > ABI.getPointerSize()) {
+      if (ConsideredRegisterCounter != OccupiedRegisterCount) {
+        if ((OccupiedRegisterCount & 1) != 0) {
+          if (ConsideredRegisterCounter < Registers.size()) {
```

Comment:
Turns the conditions in this if in a series of `bool` variables with proper names and have a single `if`:
```c++
if (HasProperSize and IsGood and NotWrong)
```

Reply:
Done:
```cpp
  if (ABI.OnlyStartDoubleArgumentsFromAnEvenRegister()) {
    const std::uint64_t PointerSize = ABI.getPointerSize();
    bool MultiAligned = (Size >= PointerSize && Alignment > PointerSize);
    bool LastRegisterUsed = ConsideredRegisterCounter == OccupiedRegisterCount;
    bool OddRegisterNext = (OccupiedRegisterCount & 1) != 0;
    if (MultiAligned && !LastRegisterUsed && OddRegisterNext) {
      // . . .
    }
  }
```